### PR TITLE
feat: add append option

### LINKED ISSUE #119: feat: Add Append to conflict resolution
**Description**

I would like to have the ability to append to a file that already exits, for example, I have a brick that adds function to a repository right now the work around that I generate temp file and move the function to the correct file then delete the temp file

**Requirements**

- [ ] Add Append options to conflict

### PR DESCRIPTION
**Pull request by MohamedAbdallah-14**
# Description

Added the append option to File Conflict Resolution
Since in conflict prompt 'Yna' you choose a (Always) I had to change it to (YnaYA) a (append) and Added YA (Yes Always) to the options

Fixes #119 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Added test to make_test.dart -> 'generates greeting and append to existing file'

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by MohamedAbdallah-14** [comment]
I have removed the pull request template since you have added one 😁

**Comment by felangel** [comment]
> I have removed the pull request template since you have added one 😁

Thanks! Will review asap!

**Comment by MohamedAbdallah-14** [comment]
Okay thanks, and let me know if the prompt 'YnaYA' is acceptable or you have another thing in mind.

File: test/commands/make_test.dart [hunk 1]
```diff
@@ -469,5 +471,45 @@ in todos.json''',
         () => logger.delayed(any(that: contains('(new)'))),
       ).called(1);
     });
+
+    test('generates greeting and append to existing file', () async {
```
Comment (felangel) [inline test/commands/make_test.dart:475]:
```suggestion
    test('generates greeting and appends to existing file', () async {
```

File: README.md [hunk 2]
```diff
@@ -102,12 +102,13 @@ $ mason make hello --name Felix -o ./path/to/directory
...
+By default, `mason make` will prompt on each file conflict and will allow users to specify how the conflict should be resolved via `YnaYA`:
 
 ```txt
-Y - overwrite (default)
-n - do not overwrite
-a - overwrite this and all others
+Y(Yes) - overwrite (default)
+n(no) - do not overwrite
+a(append) - append to existing file
+YA(Yes Always) - overwrite this and all others
```
Comment (felangel) [inline README.md:111]:
```suggestion
y - yes, overwrite (default)
Y - yes, overwrite this and all others
n - no, do not overwrite
a - append to existing file
```

What are your thoughts on ☝️ ?

File: README.md [hunk 3]
```diff
@@ -102,12 +102,13 @@ $ mason make hello --name Felix -o ./path/to/directory
 
 ### File Conflict Resolution
 
-By default, `mason make` will prompt on each file conflict and will allow users to specify how the conflict should be resolved via `Yna`:
+By default, `mason make` will prompt on each file conflict and will allow users to specify how the conflict should be resolved via `YnaYA`:
```
Comment (felangel) [inline README.md:105]:
```suggestion
By default, `mason make` will prompt on each file conflict and will allow users to specify how the conflict should be resolved via `Yyna`:
```

File: lib/src/generator.dart [hunk 4]
```diff
@@ -493,6 +510,9 @@ extension on String {
       case 'n':
         return OverwriteRule.skipOnce;
       case 'a':
+        return OverwriteRule.appendOnce;
+      case 'ya':
+      case 'YA':
```
Comment (felangel) [inline lib/src/generator.dart:515]:
```suggestion
      case 'Y':
```

**Comment by MohamedAbdallah-14** [comment]
Thanks for the review I like yYna better

**Comment by MohamedAbdallah-14** [comment]
I have fixed the analyze issue

File: lib/src/generator.dart [hunk 5]
```diff
@@ -211,11 +214,17 @@ enum OverwriteRule {
...
+  /// Always append the existing file.
+  alwaysAppend,
+
   /// Overwrite one time.
   overwriteOnce,
 
   /// Do not overwrite one time.
   skipOnce,
+
+  /// Append One time
```
Comment (felangel) [inline lib/src/generator.dart:226]:
```suggestion
  /// Append one time
```

File: lib/src/generator.dart [hunk 6]
```diff
@@ -276,13 +286,18 @@ class DirectoryGeneratorTarget extends GeneratorTarget {
         return file;
       case OverwriteRule.alwaysOverwrite:
       case OverwriteRule.overwriteOnce:
-      default:
+      case OverwriteRule.appendOnce:
+      case OverwriteRule.alwaysAppend:
+      case null:
```
Comment (felangel) [inline lib/src/generator.dart:291]:
```suggestion
      default:
```

**Review by felangel — APPROVED** [review]
LGTM 🎉 
Thanks so much for the contribution! 💙

**Pull request merged**